### ISSUE TITLE
Mention: suggestion panel out of window boundary

### ISSUE THREAD

**Issue by lagroms** [primefaces/primereact#4196]
### Describe the bug

![Screenshot 2023-03-29 at 9 46 23 AM](https://user-images.githubusercontent.com/88441919/228439058-bd8abf57-b4b2-4cd8-86d4-b191d284e167.png)

In the case of a chat input at the bottom of the window, the suggestion panel won't automatically relocate itself. 
I've also tried to play with the panelStyle prop, but it doesn't seem to work ? 

### PrimeReact version

9.2.1

### React version

17.x

### Language

ES6

### Build / Runtime

Create React App (CRA)

**Comment by melloware** [comment]
Confirmed this popup is not doing collision detection like AutoComplete and other components are doing its always placing it in the same spot.

```js
        const alignOverlay = () => {
            const { key, index } = triggerState;
            const value = inputRef.current.value;
            const position = DomHandler.getCursorOffset(inputRef.current, value.substring(0, index - 1), value.substring(index), key);

            overlayRef.current.style.transformOrigin = 'top';
            overlayRef.current.style.left = `calc(${position.left}px + 1rem)`;
            overlayRef.current.style.top = `calc(${position.top}px + 1.2rem)`;
        };
```

**Comment by lagroms** [comment]
Any news for this feature ?

**Comment by lagroms** [comment]
@melloware / @thejus557 any news regarding this feature?

**Comment by melloware** [comment]
I have not investigated it further only confirmed that it was an issue.

**Comment by jurvi** [comment]
Yeah this is definitely an issue, just bumped into this as well. When the Mention component is in a sidepanel opened from right side of the screen, the suggestion panel often pushes the whole viewport to right causing a horizontal scrollbar.

**Comment by briannezhad** [comment]
Is there a solution for this? I wish there were a additional prop for positioning

**Comment by melloware** [comment]
no i think the core component would have to be modified.

**Comment by potatosalad94** [comment]
@cagataycivici mentioned a new generic positioning back in Feb on the discord. Don't know if it's still planned?

<img width="990" alt="Screenshot 2024-05-20 at 10 29 56" src="https://github.com/primefaces/primereact/assets/44224578/5ff69362-c12a-4278-9315-c43238f461bd">

**Comment by melloware** [comment]
@potatosalad94 yes for v11
